logger.go

```diff
@@ -124,6 +100,77 @@ func (l *Logger) logf(format string, args ...interface{}) {
 	l.lock.Unlock()
 }
 
+// calldepth 0 identifying the caller of reportCaller()
+func (l *Logger) reportCaller(calldepth int) string {
+	if !(l.callerFile || l.callerFunc || l.callerPkg) {
+		return ""
+	}
+
+	filePath, line, funcName := l.caller(calldepth + 1)
+
+	// callerPkg only if no other callers
+	if l.callerPkg && !l.callerFile && !l.callerFunc {
+		if filePath == "" {
```

Comment:
is this (and 2 others) are needed? In what case it may be empty? I see caller may return "", but can't imagine in what sistuation. I don't see covarage for such cases either

Reply:
> In what case it may be empty?

`filePath` and `funcName` may be empty strings and `line` may be zero if
- [`runtime.Callers()`](https://golang.org/pkg/runtime/#Callers) can't read the requested program counter
- frame, returned by [`runtime.CallersFrames()`](https://golang.org/pkg/runtime/#CallersFrames) doesn't contain such information ([see docs](https://golang.org/pkg/runtime/#Frame))

> I see caller may return "", but can't imagine in what sistuation.

I also can't imagine this situation. It is like error checking. If we aim to write robust programs then we should check for errors.

> I don't see covarage for such cases either

I'm not sure whether to do it. As you can see this case is rather theoretical. The similar case in the `master` branch is also not tested (if `ok == false`):

https://github.com/go-pkgz/lgr/blob/e8926afdb882410a402a69e845d43d5554a414a9/logger.go#L78

Reply:
> I also can't imagine this situation. It is like error checking. If we aim to write robust programs then we should check for errors.

AFAICS this is not exactly an accurate analogy.  "Caller" doesn't return an error, but ok flag. The same story with the new call chain. I'd say - if you think this is an error and caller code has to treat it as an error - just return error. Otherwise returned value should be safe to use and frankly I don't see a particular reason why "???" any better indication of hypothetical lack of caller info compared to the default value ("") for the end user.

However, with your last commit this issue doesn't bother me too much anymore, so if you want to keep this ""->"???" transformation once – fine with me.

Reply:
> if you think this is an error

I don't have any particular opinion on this case. The only thing I try to achieve is to log something distinguishable to indicate the lack of the caller info. The problem that the rest of the code may not handle this case properly. For example if `filePath == ""` then pkgInfo will be `"."` which may confuse the user.

What you propose to do? Remove the check for the lack of the caller info?

Reply:
I can live with a single extra if